### PR TITLE
Added PUSHA,PUSHAD,POPA,POPAD x86 instructions

### PR DESCRIPTION
**Pull request by ramiscoding**
Test cases for these instructions are being worked on. The auto testcase generation tool doesn't handle these instructions.

### CODE REVIEW

File: manticore/core/cpu/x86.py [hunk 1]
```diff
@@ -3067,6 +3067,78 @@ def PUSHFQ(cpu):
...
+
+        The order in which the register values are pushed into the stack
+        is the same as mentioned in the list below.The value of SP pushed
+        is the value prior to the first push.
+
+        :param cpu: current CPU.
+        '''
+        temp_SP = cpu.read_register('SP')
+        registers = ['AX', 'CX', 'DX', 'BX', temp_SP, 'BP', 'SI', 'DI']
+        for reg in registers:
```
Comment (yan) [inline manticore/core/cpu/x86.py]: This code reads the value of `SP` and uses it as register name, probably not what you intended. Ditto for `PUSHAD`.

**Comment by pgoodman** [comment]
I think that the code would be simpler if you manually wrote out the actual operations on the various registers/stack, as opposed to using a loop.

File: manticore/core/cpu/x86.py (hunk 1 above)
Comment (ramiscoding) [reply manticore/core/cpu/x86.py]: That was intended behavior. The value of SP pushed into the stack should be the original value of SP before the instruction is executed. That is why I saved the SP value into another variable and pushed it to the stack.

Comment (yan) [reply manticore/core/cpu/x86.py]: The original value of SP isn't what's being pushed however. Try to write a simple test case to confirm these semantics.

**Comment by mossberg** [comment]
hi @ramiscoding! since it looks like progress on this has slowed, i am going to close this pr for the time being. feel free to reopen if you look into it again, and of course feel free to ask any questions for clarification :)

**Pull request closed without merging**